.devcontainer/devcontainer.json

```diff
@@ -1,5 +1,30 @@
 {
-  "image": "mcr.microsoft.com/devcontainers/universal:2",
+  "dockerComposeFile": "codespaces-compose.yaml",
+  "service": "main",
+  "runServices": [
+    "db",
+    "rabbitmq"
+  ],
+  "forwardPorts": [
+    8000,
+    13306,
+    15672
+  ],
+  "portsAttributes": {
+    "8000": {
+      "label": "Application"
+    },
+    "13306": {
+      "label": "MariaDB",
+      "onAutoForward": "notify"
+    },
+    "15672": {
+      "label": "RabbitMQ",
+      "onAutoForward": "notify"
+    }
+  },
+  "postCreateCommand": "cp .env.devcontainer .env && just dev_setup_codespaces || echo '.env.devcontainer not found'",
```

Comment:
We have this defined in two places because we want to run this when we rebuild a devcontainer as well as when we create it for theh first time.

Reply:
Just two things that might simplify this:

- Do we need the error check at the end? `.env.devcontainer` is tracked in git so should always be present, no?
- Could (and should?) we move the copying of the env file _into_ the just `dev_setup_codespaces` task so that each of the commands defined here can just call out to just, and all the details are kept in a single place?

If we do that (and possibly also if we don't), it might also be a good idea to have the command fail rather than overwrite an existing .env - give your personal .env isn't checked in and might have a bunch of hard-to-remember customisations, we don't want to make it too easy to overwrite accidentally.